apps/next/src/app/pages/next-index-page/next-index-page.module.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { Route, RouterModule } from '@angular/router';
+import { NextIndexPage } from './next-index-page';
+import { DtThemingModule } from '@dynatrace/barista-components/theming';
+import { BaPageGuard } from '@dynatrace/shared/data-access-strapi';
+
+export const routes: Route[] = [
```

Comment:
That is copied from barista but I do not understand why we have to define this route with an empty path here. Can someone elaborate on that?

Reply:
This is a module that includes routing information - with the RouterModule.forChild(routes) you can configure the route tree starting with this module. 

Think about your app routing - lets say you have the following route configured
/index

All components for index are loaded as a separate IndexModule. The index module also defines potential subroutes

So in the root routing config you would specify `{ path: 'index' loadChildren: import(./index.module).then(...) }`. Once the user navigates to /index the index module is loaded and it contains subroutes one of them is an empty path. Which means that whenever /index is called - you need to serve this as your component for that specific route. If you would use another string here - lets say 'abc' for the path - the new route would be /index/abc. The routes are combined starting from the root routing module to the child routing modules.

Reply:
In that case, the routes can be generated dynamically on the build. Meaning we could have a build step calculating which routes would be under /index and provide them to the index.module. Right?